gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -0,0 +1,310 @@
+    super.initialize(path, config);
+    this.instrumentation = new GhfsInstrumentation(path);
+  }
+
+  @Override
+  protected void configureBuckets(GoogleCloudStorageFileSystem gcsFs) throws IOException {}
+
+  @Override
+  public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_OPEN);
+    return super.open(hadoopPath, bufferSize);
+  }
+
+  @Override
+  public FSDataOutputStream create(
+      Path hadoopPath,
+      FsPermission permission,
+      boolean overwrite,
+      int bufferSize,
+      short replication,
+      long blockSize,
+      Progressable progress)
+      throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_CREATE);
+    FSDataOutputStream response = null;
+    try {
+      response =
+          super.create(
+              hadoopPath, permission, overwrite, bufferSize, replication, blockSize, progress);
+      instrumentation.fileCreated();
+    } catch (IOException e) {
+      throw e;
+    }
+    return response;
+  }
+
+  @Override
+  public FSDataOutputStream createNonRecursive(
+      Path hadoopPath,
+      FsPermission permission,
+      EnumSet<CreateFlag> flags,
+      int bufferSize,
+      short replication,
+      long blockSize,
+      Progressable progress)
+      throws IOException {
+
+    entryPoint(GhfsStatistic.INVOCATION_CREATE_NON_RECURSIVE);
+    return super.createNonRecursive(
+        hadoopPath, permission, flags, bufferSize, replication, blockSize, progress);
+  }
+
+  @Override
+  public boolean rename(Path src, Path dst) throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_RENAME);
+    return super.rename(src, dst);
+  }
+
+  @Override
+  public boolean delete(Path hadoopPath, boolean recursive) throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_DELETE);
+
+    boolean response = false;
+    try {
+      response = super.delete(hadoopPath, recursive);
+      incrementStatistic(
+          GhfsStatistic.OBJECT_DELETE_OBJECTS,
+          getGcsFs()
+              .getGcs()
+              .getObjectStatistics(GoogleCloudStorageStatistics.OBJECT_DELETE_OBJECTS)
+              .longValue());
+      instrumentation.fileDeleted(1);
+    } catch (IOException e) {
+      incrementStatistic(GhfsStatistic.FILES_DELETE_REJECTED);
+      throw e;
+    }
+    return response;
+  }
+
+  @Override
+  public FileStatus[] listStatus(Path hadoopPath) throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_LIST_STATUS);
+    entryPoint(GhfsStatistic.INVOCATION_LIST_FILES);
+    return super.listStatus(hadoopPath);
+  }
+
+  @Override
+  public boolean mkdirs(Path hadoopPath, FsPermission permission) throws IOException {
+    entryPoint(GhfsStatistic.INVOCATION_MKDIRS);
+    boolean response = false;
+    try {
+      response = super.mkdirs(hadoopPath, permission);
+      instrumentation.directoryCreated();
+    } catch (IOException e) {
+      throw e;
+    }
```

Comment:
Unless we capture exception in a metric we don't need a try catch block. This is a little non-performant